### PR TITLE
#517 - Add factory function

### PR DESCRIPTION
**Pull request by treshugart**
Implements #517. Already merged into #541.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1006,6 +1007,20 @@ skate.emit(element, 'event', {
 
 
 
+### `factory (componentDefinition)`
+
+The `factory()` function gives you a way to define a custom element without defining its name. This is useful because it allows your consumers to decide which name your component should have. This is also effective in maintaining UI components that may have breaking changes made to them as you don't have to change their global name in order to have multiple versions of them on the same page. It can be up to your consumers to decide how they want to do that based on how they're using them.
+
+```js
+var oldComponent = skate.factory({ ... });
```
Comment (joscha) [inline README.md]: Why not simplify this to:

```javascript
var componentFactory = skate.factory({ ... });
var myComponent = componentFactory('my-component');
```

**Comment by joscha** [comment]
LGTM

**Comment by seancurtis** [comment]
LGMT

**Comment by seancurtis** [comment]
LGTM

**Comment by jpnelson** [comment]
LGTM

File: README.md (hunk 1 above)
Comment (treshugart) [inline README.md]: Done.

**Pull request merged**